kernel/kernel-api/src/main/java/io/delta/kernel/internal/actions/Metadata.java

```diff
@@ -132,7 +138,11 @@ public Optional<Long> getCreatedTime() {
         return createdTime;
     }
 
-    public Map<String, String> getConfiguration() {
+    public MapValue getConfiguration() {
         return configuration;
     }
+
+    public String getColumnMappingMode() {
```

Comment:
hm. I don't know if this is a very scalable or future-proof design.

we won't have getters for every other configuration, will we? (I don't think so)

So, what makes this one special?

Instead, we should just lazily convert the configuration MapValue into a Map on the first `.getTableProperty(key: String)` and access it that way

Reply:
Added `getConfiguration --> Map<String, String>` instead. Let me know if you'd prefer a getTableProperty API instead, but I think we'd have to provide a couple methods
- boolean isTablePropertySet(key)
- String getTableProperty(key)
- String getTableProperty(key, default) <-- optional